examples/typescript-example/src/App.tsx

```diff
@@ -0,0 +1,226 @@
+`
+
+const TableWrap = styled.div`
+  display: inline-block;
+  border-spacing: 0;
+  border: 1px solid black;
+`
+
+const TableHead = styled.div``
+const TableBody = styled.div``
+const TableRow = styled.div``
+
+const TableCell = styled.div`
+  position: relative;
+  margin: 0;
+  padding: 0.5rem;
+  border-bottom: 1px solid black;
+  border-right: 1px solid black;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  :last-child {
+    border-right: 0;
+  }
+`
+const Resizer = styled.div`
+  display: inline-block;
+  width: 5px;
+  height: 100%;
+  position: absolute;
+  right: 0;
+  top: 0;
+  transform: translateX(50%);
+  z-index: 1;
+`
+
+const Input = styled.input`
+  width: calc(100% - 0.5rem);
+`
+
+const useStopPropagationOnClick = <T extends any>(ref: T) => {
+  useEffect(() => {
+    if (ref.current !== null) {
+      ref.current.addEventListener('click', (e: MouseEvent) => e.stopPropagation())
+    }
+  }, [])
+}
+
+const DefaultColumnFilter = ({
+  column: { filterValue, preFilteredRows, setFilter }
+}: {
+  column: TableColumn<Data>
+}) => {
+  const count = preFilteredRows.length
+  const inputRef = useRef<HTMLInputElement>(null)
+  useStopPropagationOnClick(inputRef)
+  return (
+    <Input
+      ref={inputRef}
+      value={filterValue || ''}
+      onChange={e => {
+        setFilter(e.target.value || undefined) // Set undefined to remove the filter entirely
+      }}
+      placeholder={`Search ${count} records...`}
+    />
+  )
+}
+
+const ResizerComponent: FC = props => {
+  const ref = useRef<HTMLDivElement>(null)
+  useStopPropagationOnClick(ref)
+  return <Resizer {...props} ref={ref} />
+}
+
+const defaultColumn = {
+  minWidth: 20,
+  width: 150,
+  maxWidth: 500,
+  Filter: DefaultColumnFilter
+}
+
+const Table: FC<Props> = ({ columns, data }) => {
+  // Use the state and functions returned from useTable to build your UI
+  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } = useTable<Data>(
+    {
+      columns,
+      defaultColumn,
+      data
+    },
+    useFilters,
+    useSortBy,
+    useBlockLayout,
+    useResizeColumns
+  ) as TableInstance<object>
+  // Render the UI for your table
+  return (
+    <TableWrap {...getTableProps()}>
+      <TableHead>
+        {headerGroups.map(headerGroup => (
+          <TableRow {...headerGroup.getHeaderGroupProps()}>
+            {headerGroup.headers.map(c => {
+              const column = (c as unknown) as TableColumn<Data>
+              return (
+                <TableCell {...column.getHeaderProps(column.getSortByToggleProps())}>
+                  {column.render('Header')}
+                  <OrderIndicator {...column} />
+
+                  <div>{column.canFilter ? column.render('Filter') : null}</div>
+                  <ResizerComponent {...column.getResizerProps()} />
+                </TableCell>
+              )
```

Comment:
i'm trying to add a sort icon next to the titles of the columns, but the icon also appears next to header coming from {column.render('Header')}. Could you please suggest how to not make the icon appear next to the Header? This is my code: `return (
                <TableCell key={key} {...column.getHeaderProps(column.getSortByToggleProps())}>
                  {column.render('Header')}
                  <OrderIndicator {...column} />
                  <Icon type="sort" />
                  <div>{column.canFilter ? column.render('Filter') : null}</div>
                  <ResizerComponent {...column.getResizerProps()} />
                </TableCell>
              )`

Reply:
never mind! I figured it out.